### PR TITLE
🎈 6.10.0

### PR DESCRIPTION
**Pull request by chapulina**
# 🎈 Release

Prepare for 6.10.0 release.

Comparison to 6.9.2: https://github.com/ignitionrobotics/ign-math/compare/ignition-math6_6.9.2...ign-math6

This follows the 6.9.3~pre2 pre-release:

* https://github.com/ignitionrobotics/ign-math/pull/367

Various updates have been made to packaging too:

* https://github.com/ignition-release/ign-math6-release/pull/7
* https://github.com/ignition-release/ign-math6-release/pull/8
* https://github.com/ignition-release/ign-math6-release/pull/9

I think we're in a good point to call our Python interfaces a feature, that's why I bumped the minor instead of the patch version.

Required by

* https://github.com/ignitionrobotics/ign-gazebo/pull/1219

## Checklist
- [x] Asked team if this is a good time for a release
- [x] There are no changes to be ported from the previous major version
- [x] No PRs targeted at this major version are close to getting in
- [x] Bumped minor for new features, patch for bug fixes
- [x] Updated changelog
- [x] Updated migration guide (as needed)
- [x] ~~Link to PR updating dependency versions in appropriate repository in [ignition-release](https://github.com/ignition-release) (as needed): <LINK>~~

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

**Review by ahcorde — APPROVED** [review]

**Comment by scpeters** [comment]
I just did a quick test of `ign-math6` on macOS with https://github.com/osrf/homebrew-simulation/pull/1776 and I'm seeing just one python test failure and several ruby test failures. I think it will be pretty easy to fix the python test failures and enable the `pybind11` dependency in homebrew and revisit the ruby failures later

**Comment by chapulina** [comment]
> I think it will be pretty easy to fix the python test failures and enable the pybind11 dependency in homebrew and revisit the ruby failures later

Nice! Do you want to tackle the test failure before the release? It would be great to get this release out today or tomorrow, so that we can merge the `ign-gazebo` PR by the end of the week :pray:

**Comment by scpeters** [comment]
> > I think it will be pretty easy to fix the python test failures and enable the pybind11 dependency in homebrew and revisit the ruby failures later
> 
> Nice! Do you want to tackle the test failure before the release? It would be great to get this release out today or tomorrow, so that we can merge the `ign-gazebo` PR by the end of the week 🙏

I'll open a pull request today to support pybind11

**Comment by scpeters** [comment]
> > > I think it will be pretty easy to fix the python test failures and enable the pybind11 dependency in homebrew and revisit the ruby failures later
> > 
> > 
> > Nice! Do you want to tackle the test failure before the release? It would be great to get this release out today or tomorrow, so that we can merge the `ign-gazebo` PR by the end of the week 🙏
> 
> I'll open a pull request today to support pybind11

https://github.com/ignitionrobotics/ign-math/pull/361 and https://github.com/osrf/homebrew-simulation/pull/1793

File: Changelog.md [hunk 1]
```diff
@@ -86,6 +87,7 @@
         * [Pull request #317](https://github.com/ignitionrobotics/ign-math/pull/317)
     1. Quaternion
         * [Pull request #324](https://github.com/ignitionrobotics/ign-math/pull/324)
+        * [Pull request #361](https://github.com/ignitionrobotics/ign-math/pull/361)
```
Comment (scpeters) [inline Changelog.md:90]: this is in two places?

Comment (chapulina) [reply Changelog.md:90]: Yeah because that PR affected both Quaternion and PID

**Review by scpeters — APPROVED** [review]
ok, we can go ahead and make the release, but let's wait to `build bottle` until we've merged https://github.com/osrf/homebrew-simulation/pull/1793

**Pull request merged**